packages/apollo-server-lambda/src/ApolloServer.ts

```diff
@@ -173,30 +180,86 @@ export class ApolloServer extends ApolloServerBase {
         }
       }
 
-      const callbackFilter: APIGatewayProxyCallback = (error, result) => {
-        callback(
-          error,
-          result && {
-            ...result,
-            headers: {
-              ...result.headers,
-              ...requestCorsHeaders,
+      const makeCallbackFilter = (doneFn: any): APIGatewayProxyCallback => {
+        return (error, result) => {
+          doneFn();
+          callback(
+            error,
+            result && {
+              ...result,
+              headers: {
+                ...result.headers,
+                ...requestCorsHeaders,
+              },
             },
-          },
-        );
+          );
+        };
       };
 
-      graphqlLambda(async () => {
-        // In a world where this `createHandler` was async, we might avoid this
-        // but since we don't want to introduce a breaking change to this API
-        // (by switching it to `async`), we'll leverage the
-        // `GraphQLServerOptions`, which are dynamically built on each request,
-        // to `await` the `promiseWillStart` which we kicked off at the top of
-        // this method to ensure that it runs to completion (which is part of
-        // its contract) prior to processing the request.
-        await promiseWillStart;
-        return this.createGraphQLServerOptions(event, context);
-      })(event, context, callbackFilter);
+      const response = new Stream.Writable();
+
+      fileUploadProcess(event, response, this.uploadsConfig || {})
```

Comment:
The code here would seem to work, but this is a confusing arrangement of logic.  The `fileUploadProcess` shouldn't be the thing that kicks off the chain, it should be an optional chain within the path of request handling.

To put it another way, let's maintain that `graphqlLambda` is the thing that gets kicked off, and the callback passed to `graphqlLambda` be the part of the code that brings the upload processing`, rather than having `graphqlLambda` triggered within `fileUploadProcess` — that's not an intuitive place for it.